### ISSUE TITLE
[BUG] `deepfreeze` modifies the original `dict`

### ISSUE THREAD

**Issue by kenodegard** [Marco-Sulla/python-frozendict#96]
Welcome! You should write in your Bug Report:

OS version (https://www.google.com/search?channel=fs&q=check+os+version&ie=utf-8&oe=utf-8):
macOS 14.3

Python3 version (python3 -V -V):
Python 3.12.2

Steps to reproduce:

1. Create some nested dict and call deepfreeze on it.

Actual result (with the python stack trace if present):

```pycon
>>> o = {'a': 1, 'nested': {'b': 2}}
>>> d = deepfreeze(o)
>>> d
frozendict.frozendict({'a': 1, 'nested': frozendict.frozendict({'b': 2})})
>>> o
{'a': 1, 'nested': frozendict.frozendict({'b': 2})}
```

---

I expected `deepfreeze` to leave the original object untouched.

**Comment by Marco-Sulla** [comment]
Well, that's why is called *deep*freeze. If you want to freeze only the main object, do

```
d = frozendict.frozendict(o)
```

**Comment by kenodegard** [comment]
But `frozendict` will only make the outter most object immutable.

Isn't the goal of `deepfreeze` to create a recursively immutable object?

I'm not understanding why `deepfreeze` should have any affect the original data structure.

**Comment by Marco-Sulla** [comment]
> I'm not understanding why `deepfreeze` should have any affect the original data structure.

Can you please provide the expected output?

**Comment by Marco-Sulla** [comment]
Sorry for misunderstanding, the patch will be released in v2.4.2
